bc/sponsorship/models.py

```diff
@@ -0,0 +1,79 @@
+from django.db import models
+
+from bc.core.models import AbstractDateTimeModel
+from bc.users.models import User
+
+
+class Sponsorship(AbstractDateTimeModel):
+    user = models.ForeignKey(
+        User,
+        help_text="The user sponsoring the Bot",
+        related_name="sponsorships",
+        on_delete=models.CASCADE,
+    )
+    original_amount = models.DecimalField(
+        help_text="Initial amount of money given by the sponsor",
+        max_digits=10,
+        decimal_places=2,
+    )
+    current_amount = models.DecimalField(
+        help_text="Amount of money that's left to buy documents",
+        max_digits=10,
+        decimal_places=2,
+        default=0,
+    )
+
+    class Meta:
+        ordering = ["date_created"]
+
+    def __str__(self) -> str:
+        return f"{self.pk}: {self.user.username} - {self.original_amount}"
+
+
+class Transaction(AbstractDateTimeModel):
+    SPONSORSHIP = 1
+    DOCUMENT_PURCHASE = 2
+    ADJUSTMENT = 3
+    CHOICES = (
+        (SPONSORSHIP, "Sponsorship"),
+        (DOCUMENT_PURCHASE, "Document Purchase"),
+        (ADJUSTMENT, "Adjustment"),
+    )
+
+    user = models.ForeignKey(
+        User,
+        help_text="User associated with the transaction",
+        related_name="transactions",
+        on_delete=models.CASCADE,
```

Comment:
We shouldn't delete transactions. Instead of CASCADE, how about PROTECT? 

That should prevent the user from being deleted if they have transaction records. In that case, we'd want to keep the user around, but set the user as a stub account or similar.

Reply:
Thanks for the suggestion. I'll use `PROTECT` instead.